lane.py

```diff
@@ -100,6 +103,69 @@ def __init__(self, entrypoint=None, **kwargs):
         """
         self.entrypoint = entrypoint
 
+    @classmethod
+    def from_request(
+            cls, library, facet_config, get_argument, worklist, **extra_kwargs
+    ):
+        """Load a faceting object from an HTTP request.
+
+        :param facet_config: A Library (or mock of one) that knows
+           which subset of the available facets are configured.
+
+        :param get_argument: A callable that takes one argument and
+           retrieves (or pretends to retrieve) a query string
+           parameter of that name from an incoming HTTP request.
+
+        :param worklist: A WorkList associated with the current request,
+           if any.
+
+        :param extra_kwargs: A dictionary of keyword arguments to pass
+           into the constructor when a faceting object is instantiated.
+
+        :return: A FacetsWithEntryPoint, or a ProblemDetail if there's
+            a problem with the input from the request.
+        """
+        return cls._from_request(
+            facet_config, get_argument, worklist, **extra_kwargs
+        )
+
+    @classmethod
+    def _from_request(
+            cls, facet_config, get_argument, worklist, **extra_kwargs
+    ):
+        """Load a faceting object from an HTTP request.
+
+        Subclasses of FacetsWithEntryPoint can override `from_request`,
+        but call this method to load the EntryPoint and actually
+        instantiate the faceting class.
+        """
+        entrypoint_name = get_argument(
+            Facets.ENTRY_POINT_FACET_GROUP_NAME, None
+        )
+        entrypoint = cls.load_entrypoint(entrypoint_name, worklist)
+        if isinstance(entrypoint, ProblemDetail):
+            return entrypoint
+        return cls(entrypoint=entrypoint, **extra_kwargs)
+
+    @classmethod
+    def load_entrypoint(cls, name, worklist):
+        """Look up an EntryPoint by name, assuming it's allowed in the
+        given WorkList.
+
+        :return: An EntryPoint class. This will be the requested
+        EntryPoint if possible. If a nonexistent or unusable
+        EntryPoint is requested, the WorkList's default EntryPoint
+        will be returned. If the WorkList has no EntryPoints, or no
+        WorkList is provided, None will be returned.
+        """
+        if not worklist or not worklist.entrypoints:
+            return None
+        default = worklist.entrypoints[0]
+        ep = EntryPoint.BY_INTERNAL_NAME.get(name)
+        if not ep or ep not in worklist.entrypoints:
+            return default
```

Comment:
are there any cases that should return a problem detail?

Reply:
Not right now, I'm leaving that space for subclasses that do weird things with entry points.